### PR TITLE
Add NFT Support

### PR DESCRIPTION
**Pull request by h0ngcha0**
### Issue / feature description

Support NFT

### Changes

- Display NFT collection
- Support transferring of the NFT

### Checklist

- [ ] Rebased to the last commit of the target branch (or merged)
- [x] Code self-reviewed
- [x] Code self-tested
- [ ] Tests updated (if needed)
- [ ] All tests are passing locally

Please keep your pull request as small as possible. If you need to make multiple changes, please create separate pull requests for each change. Create a draft pull request if you need early feedback. This will mark the pull request as a work in progress and prevent it from being reviewed or merged until you are ready.

Please move drafts to the ready for review (regular PR) when you are ready to start the review process and other developers will pick it up from there.

### CODE REVIEW

**Review by Lbqds — COMMENTED** [review]
Great work 👍 , only a few comments and questions.

File: packages/extension/src/ui/features/accountNfts/alephium-nft.service.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+async function getCollectionMetadata(
+  collectionAddress: string,
+  nodeProvider: NodeProvider
+) {
+  const collectionState = await nodeProvider.contracts.getContractsAddressState(
+    collectionAddress,
+    { group: groupOfAddress(collectionAddress) }
+  )
+
+  const metadataUri = hexToString((collectionState.immFields[1] as ValByteVec).value)
```
[the remark this answers is not in the document]
Comment (Lbqds) [reply packages/extension/src/ui/features/accountNfts/alephium-nft.service.ts:135]: I found that we already [have a function](https://github.com/alephium/alephium-web3/blob/7c59d1694b4205fd169f4a5cf4eb14b4401a0e46/packages/web3/src/api/node-provider.ts#L127) for retrieving NFT metadata, but we don't have a function for retrieving NFT collection metadata.

[the remark this answers is not in the document]
Comment (h0ngcha0) [reply packages/extension/src/ui/features/accountNfts/alephium-nft.service.ts:135]: Yes, I will add one in web3 :+1:

File: packages/extension/src/ui/features/accountNfts/SendNftScreen.tsx [hunk 2]
```diff
@@ -87,271 +82,250 @@ export const SendNftSchema: Schema<SendNftInput> = object().required().shape({
...
+    defaultValues: {
+      recipient: "",
+    },
+    resolver,
+  })
+
+  const formValues = watch()
+  const inputRecipient = formValues.recipient
+
+  const validateStarknetAddress = useCallback(
```
[the remark this answers is not in the document]
Comment (h0ngcha0) [reply packages/extension/src/ui/features/accountNfts/SendNftScreen.tsx:140]: Great catch :+1:

File: packages/extension/src/ui/features/accountTokens/tokens.state.ts [hunk 3]
```diff
@@ -165,11 +165,61 @@ export const useTokensWithBalance = (
...
+      const foundIndex = knownTokensInNetwork.findIndex((token) => token.id == userToken.id)
+      if (foundIndex !== -1) {  // in the known token list
+        result.push([knownTokensInNetwork[foundIndex], foundIndex])
+      } else if (account?.networkId === 'devnet') {
+        result.push([devnetToken(userToken), knownTokensInNetwork.length])
+      }
+    }
+  }
+
+  return result.sort((a, b) => a[1] - b[1]).map(tuple => tuple[0])
```
[the remark this answers is not in the document]
Comment (h0ngcha0) [reply packages/extension/src/ui/features/accountTokens/tokens.state.ts]: Use `_.sortBy` now :+1:

File: packages/extension/src/ui/features/accountNfts/alephium-nft.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lbqds) [reply packages/extension/src/ui/features/accountNfts/alephium-nft.service.ts]: I added one here: https://github.com/alephium/alephium-web3/pull/195

**Comment by polarker** [comment]
@h0ngcha0 Quick question: do we add extra data table for this new feature?

**Comment by h0ngcha0** [comment]
> @h0ngcha0 Quick question: do we add extra data storage for this new feature?

No extra data storage, images are loaded from IPFS.

**Review by Lbqds — APPROVED** [review]
LGTM 👍

**Comment by polarker** [comment]
I don't have the time to test&review, so let's merge once you guys think it's ready

**Comment by h0ngcha0** [comment]
> 0.8.0 LGTM in case my reply got lost on Slack.

Got it!

**Pull request merged**